### PR TITLE
Dashed text (#8)

### PR DESCRIPTION
**Pull request by danya02**

### CODE REVIEW

File: translation.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ March 2005
 Recently I've had several emails from computer science undergrads asking what to do in college. I might not be the best source of advice, because I was a philosophy major in college. But I took so many CS classes that most CS majors thought I was one. I was certainly a hacker, at least.
 
 Hacking
--------
+–––––––
```
Comment (wldhx) [inline translation.rst]: Shouldn't be changed.

File: translation.rst [hunk 2]
```diff
@@ -18,13 +18,13 @@ Odds are this project won't be a class assignment. My friend Robert learned a lo
...
 
-Some of the smartest people around you are professors. So one way to find interesting work is to volunteer as a research assistant. Professors are especially interested in people who can solve tedious system-administration type problems for them, so that is a way to get a foot in the door. What they fear are flakes and resume padders. It's all too common for an assistant to result in a net increase in work. So you have to make it clear you'll mean a net decrease.
+Some of the smartest people around you are professors. So one way to find interesting work is to volunteer as a research assistant. Professors are especially interested in people who can solve tedious system–administration type problems for them, so that is a way to get a foot in the door. What they fear are flakes and resume padders. It's all too common for an assistant to result in a net increase in work. So you have to make it clear you'll mean a net decrease.
```
Comment (wldhx) [inline translation.rst]: Shouldn't be changed.

**Comment by wldhx** [comment]
I believe you'd be better off rebasing than merging here - this commit history is quite a mess.

Also, simple `s/-/–/g` won't work properly: see the comments for reference.

**Comment by danya02** [comment]
OK, will fix those points. As for rebasing, the only thing I know of it is that such a function exists. Could you please point me to where I can find out how to perform such an operation?

**Comment by wldhx** [comment]
Welp, `man git-rebase` might be a start :)

In this case the simplest thing to do would be fetch the upstream state, branch from it, and cherry-pick the useful commits (17beb5c and f973ef3).

In the future, when the upstream has updated and you want to base your changes on its newer version, fetch it, and rebase your changes branch on it. (As simple as `git rebase remote_branch`!)

**Comment by danya02** [comment]
Tried to do it like you said, but I'm _guessing_ I made a mistake somewhere.

**Comment by wldhx** [comment]
Hard to debug from here, but I'm guessing you didn't create an entirely new branch from upstream and populate it via cherry-picks, rather did something else.

**Comment by danya02** [comment]
Maybe. Will try to fix it later today.

**Comment by wldhx** [comment]
You could also try rebasing even now: `git rebase $upstream_master` (look up `$upstream_master` via `git branch -a`).
(And for reference, when I say "upstream", I mean https://github.com/goto-ru/paulgraham-college, not your fork)

**Comment by danya02** [comment]
I can't find the upstream branch in `git branch -a` output. It seems to me this has all gotten very confusing, so here's an idea: I could remove this branch, create a new one, commit the relative changes into it and make a new pull request. (Yes, I know this is a very losing method, but it seems to be the simplest of those that we have.)

**Comment by wldhx** [comment]
I'm afraid this won't exactly fix the actual problem... (unless you are proposing to re-fork)

You should get the current version of the project, and here's what you need to do:
- add upstream as remote (`git remote add upstream git@github.com:goto-ru/paulgraham-college.git`)
- fetch from it (`git fetch upstream`)
- rebase as specified higher

This should help, ping me if it doesn't.

**Comment by danya02** [comment]
Well, I did like you told me, but the latest version doesn't incorporate the changes that we're trying to make. Is that how it was supposed to be, or is there some kind of mistake?

**Comment by wldhx** [comment]
It is a feature: you have your changes, but you want to base them on a newer version.

**Comment by danya02** [comment]
This time I did the change manually, substituting only the minuses with spaces surrounding them. But it doesn't seem like the commit history improved at all. Hopefully you have squash merging enabled!

**Comment by wldhx** [comment]
@danya02 So this PR is ready to be merged, right?

**Comment by danya02** [comment]
One moment, let me check with a hex editor...

File: translation.rst [hunk 3]
```diff
@@ -139,7 +139,7 @@ Notes
 .. [5] А если хотите, чтобы была идеальная жизнь, то стоит поступить в аспирантуру, тайно написать диссертацию в первые два года, а затем развлекаться следующие 3 года, записывая по главе за раз. Аспиранты позавидовали бы идее, но ни у кого из тех, кого я знаю, не хватило на это дисциплины.
 
 .. [6]
-        Один друг-профессор сказал, что 15-20% аспирантов имеют "низкую вероятность." Под этим он имел в виду, что это люди с анкетами, идеальными во всем, за исключением того, что никто из профессоров не знает тех, кто написал рекоммендации.
+        Один друг-профессор сказал, что 15–20% аспирантов имеют "низкую вероятность." Под этим он имел в виду, что это люди с анкетами, идеальными во всем, за исключением того, что никто из профессоров не знает тех, кто написал рекоммендации.
```
Comment (danya02) [inline translation.rst:142]: `... 15–20% ...` - `E2` - en dash

File: translation.rst [hunk 4]
```diff
@@ -35,15 +35,15 @@ If I were in college now I'd probably work on graphics: a network game, for exam
...
 
 Как и многих людей, меня неправильно учили математике в детстве. Меня научили относиться к математике как к коллекции формул, которые скорее красиво выглядели, чем имели какое-либо отношение к реальной жизни (несмотря на попытки их перевода в 'word problems'), но которые нужно помнить, чтобы хорошо писать тесты.
 
-Одна из наиболее ценных вещей, которые вы будете делать в колледже - изучение того, чем является математика на самом деле. Это будет нелегко, так как многие хорошие математики плохие учителя. И хотя есть есть много популярных книг по математике, немногие кажутся хорошими. Лучшие, по моему мнению - В.В. Сойера. И, конечно, Евклид. [4]_
+Одна из наиболее ценных вещей, которые вы будете делать в колледже – изучение того, чем является математика на самом деле. Это будет нелегко, так как многие хорошие математики плохие учителя. И хотя есть есть много популярных книг по математике, немногие кажутся хорошими. Лучшие, по моему мнению – В.В. Сойера. И, конечно, Евклид. [4]_
```
Comment (danya02) [inline translation.rst:46]: `... е – и ...` -`E2` - en dash.

File: translation.rst [hunk 5]
```diff
@@ -58,7 +58,7 @@ If I were in college now I'd probably work on graphics: a network game, for exam
 
 Социология, надо сказать, тоже совершенно бесполезна, потому что завязана на нынешних общественных  `устоях  <http://www.paulgraham.com/say.html>`__. Когда 100 лет назад встречались физики, они узнавали друг от друга что-нибудь новое, когда встречались психологи, они просто начинали спорить о ценностях. Конечно они что-то от этого, да получали, но можно с уверенностью сказать, что это что-то в разы меньше того, что ты получаешь в других областях науки.
```
Comment (danya02) [inline translation.rst:59]: `... что-то ...` - `2D` - minus.

**Comment by danya02** [comment]
Is that correct? If it isn't, please point out where it's not.

**Comment by wldhx** [comment]
The idea is correct, but you've missed quite a few. What are the downsides of a simple `sed s/\ -\ /\ –\ /g translation.rst -i` once again?

**Comment by danya02** [comment]
The downside is that I don't have the brainwidth to both translate the text and remember how to make `sed` scripts. Thanks for noting that, will now apply your patch.

**Comment by danya02** [comment]
Done. So it seems.

**Comment by wldhx** [comment]
@danya02 :+1:

**Pull request merged**